apps/examples/memory_fragmentation_test/memory_fragmentation_test.c

```diff
@@ -0,0 +1,344 @@
+};
+
+static bool memory_allocation(struct alloc_list list[], int numof_size[], int num_alloc[])
+{
+	struct alloc_list *next[MAX_SIZE_EXPONENT];
+	struct alloc_list *item;
+	int max_interval;
+	int i;
+	int interval[MAX_SIZE_EXPONENT] = {0, };
+	int cur_interval[MAX_SIZE_EXPONENT] = {0, };
+
+	/* Calculate the maximum allocation count. */
+	max_interval = numof_size[0];
+	for (i = 1; i < MAX_SIZE_EXPONENT; ++i) {
+		if (numof_size[i] > max_interval) {
+			max_interval = numof_size[i];
+		}
+	}
+
+	/* Move 'next' pointer to the end of list to add a new item from the tail */
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		next[i] = &list[i];
+		while (next[i]->next) {
+			next[i] = next[i]->next;
+		}
+	}
+
+	/* Calculate the interval of allocation per size. */
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		if (numof_size[i] > 0) {
+			interval[i] = cur_interval[i] = max_interval / numof_size[i];
+		} else if (numof_size[i] == 0) {
+			interval[i] = cur_interval[i] = max_interval + 1;
+		} else {
+			printf("Invalid number of memory allocation with size %d\n", (1 << (i + 4)));
+			printf("It should be a positive number.\n");
+			return false;
+		}
+	}
+
+	/* Allocating each memory segment whenever its interval is zero */
+	while (max_interval > 0) {
+		for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+			if (cur_interval[i] == 0) {
+				if (num_alloc[i] >= numof_size[i]) {
+					cur_interval[i] = interval[i] - 1;
+					continue;
+				}
+					
+				item = (struct alloc_list *)malloc(sizeof(struct alloc_list));
+				if (item) {
+					item->data = (char *)malloc((1 << (i + 4) * sizeof(char)));
+					/* add a new item at the tail */
+					if (item->data) {
+						item->next = NULL;
+						item->prev = next[i];
+						next[i]->next = item;
+						next[i] = item;
+						++num_alloc[i];
+					} else {
+						free(item);
+						printf("Size %d's %d-th allocation failed.\n", 1 << (i + 4), num_alloc[i]);
+						printf("Too many memory allocations were tried.\n");
+						return false;
+					}
+				} else {
+					printf("Size %d's %d-th allocation failed.\n", 1 << (i + 4), num_alloc[i]);
+					printf("Too many memory allocations were tried.\n");
+					return false;
+				}
+				cur_interval[i] = interval[i] - 1;
+			} else {
+				--cur_interval[i];
+			}
+		}
+		--max_interval;
+	}
+
+	return true;
+}
+
+static bool memory_free(int num[], int num_alloc[], struct alloc_list list[])
+{
+	struct alloc_list *temp;
+	int num_free[MAX_SIZE_EXPONENT];
+	int i;
+
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		num_free[i] = num[i];
+	}
+
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		while (num_free[i]) {
+			temp = list[i].next;
+			if (temp) {
+				temp->prev->next = temp->next;
+				if (temp->next) {
+					temp->next->prev = temp->prev;
+				} 
+				free(temp->data);
+				free(temp);
+				--num_alloc[i];
+			} else {
+				return false;
+			}
+			--num_free[i];
+		}
+	}
+
+	return true;
+}
+
+static void memory_cleanup(int num_alloc[], struct alloc_list list[]) {
```

Comment:
Could you locate `{` at next line to remove a violation of coding rules?

Reply:
Thanks for your comments. It's done.